contracts/SimpleQuests.sol

```diff
@@ -0,0 +1,422 @@
+    }
+
+    function getCharacterQuestData(uint256 characterID) public view returns (uint256[] memory) {
+        uint256[] memory questDataKeys = new uint256[](3);
+        questDataKeys[0] = NFTVAR_SIMPLEQUEST_PROGRESS;
+        questDataKeys[1] = NFTVAR_SIMPLEQUEST_TYPE;
+        questDataKeys[2] = NFTVAR_REPUTATION;
+        return characters.getNFTVars(characterID, questDataKeys);
+    }
+
+    function getCharacterQuestDataDetails(uint256 characterID) public view returns (uint256[] memory, uint256, Rarity,
+        RequirementType, Rarity, uint256,
+        RewardType, Rarity, uint256, uint256) {
+        if (characterQuest[characterID] == 0) {
+            return (getCharacterQuestData(characterID), 0, Rarity.COMMON,
+            RequirementType.NONE, Rarity.COMMON, 0,
+            RewardType.NONE, Rarity.COMMON, 0, 0);
+        }
+        Quest memory quest = quests[characterQuest[characterID]];
+        return (getCharacterQuestData(characterID),
+        quest.id, quest.tier,
+        quest.requirementType, quest.requirementRarity, quest.requirementAmount,
+        quest.rewardType, quest.rewardRarity, quest.rewardAmount,
+        quest.reputationAmount);
+    }
+
+    function canSkipQuest(uint256 characterID) public view returns (bool) {
+        return characters.getStaminaPoints(characterID) >= vars[VAR_SKIP_QUEST_STAMINA_COST];
+    }
+
+    // ADMIN
+
+    function setVar(uint256 varField, uint256 value) external restricted {
+        vars[varField] = value;
+    }
+
+    function setVars(uint256[] calldata varFields, uint256[] calldata values) external restricted {
+        for (uint i = 0; i < varFields.length; i++) {
+            vars[varFields[i]] = values[i];
+        }
+    }
+
+    function setTierChances(uint256 tier, uint256[4] memory chances) public restricted {
+        tierChances[tier] = chances;
+    }
+
+    function addNewQuestTemplate(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
+        RewardType rewardType, Rarity rewardRarity, uint256 rewardAmount, uint256 reputationAmount) public restricted {
+        uint256 questID = addNewQuest(tier, requirementType, requirementRarity, requirementAmount, rewardType, rewardRarity, rewardAmount, reputationAmount);
+        questTemplates[uint8(tier)].push(questID);
+    }
+
+    function addNewPromoQuestTemplate(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
+        RewardType rewardType, Rarity rewardRarity, uint256 rewardAmount, uint256 reputationAmount) public restricted {
+        uint256 questID = addNewQuest(tier, requirementType, requirementRarity, requirementAmount, rewardType, rewardRarity, rewardAmount, reputationAmount);
+        questTemplates[uint8(tier) + 10].push(questID);
+    }
+
+    function addNewQuest(Rarity tier, RequirementType requirementType, Rarity requirementRarity, uint256 requirementAmount,
```

Comment:
Eventually it will make sense to separate the tier assignment and quest creation steps, so we don't always have to make a new quest to include one in the template list, and we might want to create quests ahead of time before including them in the list.
This is for later, don't bother with it for now, just wanted to make note of it for posterity